lib/Conversion/SeqToSV/FirRegLowering.cpp

```diff
@@ -308,6 +340,18 @@ FirRegLowering::tryRestoringSubaccess(OpBuilder &builder, Value reg, Value term,
 
 void FirRegLowering::createTree(OpBuilder &builder, Value reg, Value term,
                                 Value next) {
+  // Get the fanout from this register before we build the tree. While we are
+  // creating the tree of if/else statements from muxes, we only want to turn
+  // muxes that are on the register's fanout into if/else statements. This is
+  // required to get the correct enable inference. But other muxes in the tree
+  // should be left as ternary operators. This is desirable because we don't
+  // want to create if/else structure for logic unrelated to the register's
+  // enable.
+  auto firReg = term.getDefiningOp<seq::FirRegOp>();
+  DenseSet<Operation *> regMuxFanout;
+  getForwardSliceSimple(firReg, regMuxFanout, [&](Operation *op) {
+    return op == firReg || !isa<sv::RegOp, seq::FirRegOp>(op);
```

Comment:
nit:  Probably we can ignore instances as well?

Reply:
Good point, I think that makes sense, and it seems to have a meaningful impact on performance to avoid unnecessary tracing.